src/main/java/com/nutomic/syncthingandroid/activities/ShareActivity.java

```diff
@@ -123,7 +150,34 @@ protected void onCreate(Bundle savedInstanceState) {
             if (files.size() == 1)
                 files.entrySet().iterator().next().setValue(mShareName.getText().toString());
             Folder folder = (Folder) mFoldersSpinner.getSelectedItem();
-            new CopyFilesTask(files, folder).execute();
+            String path = folder.path;
+            if (!TextUtils.isEmpty(mSubDirectoryTextView.getText())) {
+                path += mSubDirectoryTextView.getText();
+            }
```

Comment:
This seems kind of hacky, would probably be cleaner if you used `getSavedSubDirectory()`.

Reply:
I think I will just use replace, unless you think files would be better?

Reply:
Yes `File` is definitely better than using strings.